File: cpp/485. Max Consecutive Ones.cpp
/*Given a binary array, find the maximum number of consecutive 1s in this array.
Example 1:
Input: [1,1,0,1,1,1]
Output: 3
Explanation: The first two digits or the last three digits are consecutive 1s.
The maximum number of consecutive 1s is 3.*/

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        /*这道题让我们求最大连续1的个数，我们可以遍历一遍数组，用一个计数器count来统计1的个数，
        方法是如果当前数字为0，那么count重置为0，如果不是0，count自增1，然后每次更新结果ans即可*/
        if (nums.empty()) return 0;
        int ans = 0, count = 0;
        for (int n : nums){
            count = (count + 1) * n;
            ans = max(ans, count);
        }
        return ans;
    }
};

File: python/485. Max Consecutive Ones.py
"""
485. Max Consecutive Ones
Easy

Given a binary array nums, return the maximum number of consecutive 1's in the array.

Example 1:
Input: nums = [1,1,0,1,1,1]
Output: 3
Explanation: The first two digits or the last three digits are consecutive 1s. The maximum number of consecutive 1s is 3.

Example 2:
Input: nums = [1,0,1,1,0,1]
Output: 2
"""


class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        """
        Scan through nums and keep track of current length of consecutive ones. 
        If the current count is larger than the res, update the res.
        """
        res = count = 0
        for x in nums:
            if x:
                count += 1
            else:
                count = 0
            res = max(res, count)
        return res
